students/NatalieRodriguez/Lesson09/OO_Mailroom.py

```diff
@@ -0,0 +1,172 @@
+and methods to provide access to the donor specific information that is needed.
+
+
+You’ll then want a class that handles the collection of donors. This will hold all the donor objects,
+but also methods to add a new donor, search for a given donor, etc. If you want a way to save and re-load your data,
+this class would have that too.
+
+Your class for the collection of donors will also hold the code that generates reports about multiple donors.
+
+Remember that the user interaction code (anything with an input function) should be outside of these
+“logic” or “model” classes.
+
+In general, you will have a method for each of the functions in your non-OO version.
+Which class they go it will depend on whether the method only needs the information from one donor,
+or from the whole collection.
+
+Rules of thumb for where to put methods:
+
+Hopefully, once you made your code testable, all the user-interaction code (with input()) is self contained
+in functions that don’t have any logic (data manipulation) in them. If not, then this is a good time to refactor.
+If a function does something with a single donor – it should be a method in the Donor class.
+If a function works with multiple donors – it should be in the class that handles a collection of donors.
+If a function contains a call to input() – it belongs outside of the logic classes – either stand alone in
+the module (like they are already) or perhaps all in a CLI class.'''
+
+#!/usr/bin/env python3
+import sys
+from collections import defaultdict
+
+class Donor:
+
+    def __init__(self, name, donations=None):
+
+        self.name = name
+        if donations is None:
+            self.donations = []
+        else:
+            self.donations = donations
+
+    def add_donation(self, amount):
+        try:
+            self.donations.append(int(amount))
+        except ValueError:
+            print("Enter a donation amount.")
+
+    def donation_amount(self):
+        return len(self.donations)
+
+    def total_donation(self):
+        try:
+            return sum(self.donations)
+        except TypeError:
+            return self.donations
+
+    def donation_avg(self):
+        try:
+            return self.total_donation()/ self.donation_amount()
+        except TypeError:
+            return self.donations
+
+class DonorHistory:
+
+    def __init__(self, donors=None):
+        self.donor_list = donors
+
+    def thank_you(self):
+        '''user inputs donor and amounts to dict.
+        '''
+        print('Type the full donor name (Enter "list" to show all donors or enter "dash" to return to the Donor Dashboard.')
+        new_donor = input(':')
+
+        if new_donor.lower() == 'list':
+            donor_names = []
+            for donor in self.donor_list:
+                donor_names.append(donor.name)
+            print(donor_names)
+
+        elif new_donor.lower() == 'dash':
+            return
+
+        else:
+            if new_donor in self.donor_list:
```

Comment:
the simplest way to debug is to add some `print` statements 🙂 

if you print `new_donor` and `self.donor_list` you will see that new_donor is a string (name) and `self.donor_list` is a list of Donor objects. So your if statement will always evaluate to False.